lib/src/source/hosted.dart

```diff
@@ -391,7 +401,7 @@ class BoundHostedSource extends CachedSource {
         final results = <RepairResult>[];
         var packages = <Package>[];
         for (var entry in listDir(serverDir)) {
-          if (dirExists(entry)) {
+          if (_looksLikePackageDir(entry)) {
```

Comment:
This isn't technically necessary, right?

Downside is that if you repair the cache because you delete a `pubspec.yaml` that won't work anymore :see_no_evil:

Reply:
If our directory is called `.versions` or `.cache` then `listDir` won't return it, right?